### PR TITLE
allow building qt6 with no-opengl option

### PR DESCRIPTION
**Pull request by varigigi**
Building Qt6 framework enabling OpenGL on SoCs with no 3D GPU, causes invoking OpenGL SW emulation provided by MESA: under these conditions Qt6 performances are very poor.
These changes allow building Qt6 demos for iMX93 with both fsl-imx-fb and fsl-imx-xwayland with far away better performances.

### CODE REVIEW

File: meta-sdk/dynamic-layers/qt6-layer/recipes-qt/qt6/qtbase/qt6-linuxfb.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+if test -z $XDG_RUNTIME_DIR; then
+	export XDG_RUNTIME_DIR=/run/user/$(id -u)
+	if [ ! -d $XDG_RUNTIME_DIR ]; then
+		mkdir -m 700 -p $XDG_RUNTIME_DIR
+	fi
+fi
```
Comment (thochstein) [inline meta-sdk/dynamic-layers/qt6-layer/recipes-qt/qt6/qtbase/qt6-linuxfb.sh:7]: This `XDG_RUNTIME_DIR` snippet is for Wayland, I wonder why you need it here?

**Comment by thochstein** [comment]
Thanks, @varigigi, this looks good, and I'll look at integrating it for a later release.

**Comment by varigigi** [comment]
Hi @thochstein,
although you should be right, we tested it using fsl-imx-fb (no wayland at all): Qt6 demos complain anyway for the lack of this declaration.
Any roughly estimation about a possible time-frame for the integration of these changes ?
Thanks
Best Regards
Pier

**Comment by thochstein** [comment]
Hi Pier, we have a quarterly release schedule. Unfortunately, we just missed the Q1 cutoff, so it won't be until end of Q2.

**Comment by david-wolfe** [comment]
Note also that both `QMLSCENE_DEVICE=softwarecontext` and `QT_QPA_PLATFORM=wayland` environment variables are set by NXP's nxp-demo-experience Qt app in order to work around exactly this issue: [meta-nxp-demo-experience/recipes-fsl/images/imx-image-full.bbappend](https://github.com/nxp-imx-support/meta-nxp-demo-experience/blob/kirkstone/recipes-fsl/images/imx-image-full.bbappend).

**Comment by varigigi** [comment]
FYI, I'm proceeding with Qt team also.
The compile problems with qtbase are already fixed on the mainline and branch 6.5: I'm checking how to request the backport for 6.4 and 6.2.
For the problems with qtwayland, I've opened a [QTBUG-110420](https://bugreports.qt.io/browse/QTBUG-110420) and pushed the [patch](https://codereview.qt-project.org/c/qt/qtwayland/+/455223), now in review.

**Pull request closed without merging**